pymeasure/display/windows/multiplot_window.py

```diff
@@ -0,0 +1,132 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+import os
+
+import pyqtgraph as pg
+
+from .managed_window import ManagedWindowBase
+from ..widgets.multiplot_widget import MultiPlotWidget, MultiPlotResultsDialog
+from ..widgets.log_widget import LogWidget
+from ..browser import BrowserItem
+from ..Qt import QtWidgets
+from ..manager import Experiment
+from ...experiment import Results
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class MultiPlotWindow(ManagedWindowBase):
+    """
+    A window for plotting multiple graphs that share the X axis with custom Y axis
+
+    .. seealso::
+        Tutorial :ref:`tutorial-plotterwindow`
+        A tutorial and example code for using the Plotter and PlotterWindow.
+
+    Parameters for :code:`__init__` constructor.
+
+    :param procedure_class: procedure class describing the experiment (see
+        :class:`~pymeasure.experiment.procedure.Procedure`)
+    :param x_axis: the initial data-column for the x-axis of the plot
+    :param y_axis: the initial data-column for the y-axis of the plot, either a single y-axis label
+        or a list of y-axis labels for each plot
+    :param linewidth: linewidth for the displayed curves, default is 1
+    :param num_plots: number of plots to display, default is 1
+    :param \\**kwargs: optional keyword arguments that will be passed to
+        :class:`~pymeasure.display.windows.managed_window.ManagedWindowBase`
+
+    """
+
+    def __init__(self, procedure_class, x_axis=None, y_axis=None, linewidth=1, num_plots=1,
+                 **kwargs):
+        self.x_axis = x_axis
+        self.y_axis = y_axis
+        self.num_plots = num_plots
+        self.log_widget = LogWidget("Experiment Log")
+        self.plot_widget = MultiPlotWidget("Results Graph", procedure_class.DATA_COLUMNS, x_axis,
+                                           y_axis,
+                                           num_plots=num_plots, linewidth=linewidth)
+        self.plot_widget.setMinimumSize(100, 200)
+
+        if "widget_list" not in kwargs:
+            kwargs["widget_list"] = ()
+        kwargs["widget_list"] = kwargs["widget_list"] + (self.plot_widget, self.log_widget)
+
+        super().__init__(procedure_class, **kwargs)
+
+        # Setup measured_quantities once we know x_axis and y_axis
+        measure_quantities = [self.x_axis, self.y_axis]
+        if type(self.y_axis) == list:
+            # Expand y_axis if it is a list
+            measure_quantities = [self.x_axis, *self.y_axis]
+        self.browser_widget.browser.measured_quantities = measure_quantities
+
+        logging.getLogger().addHandler(self.log_widget.handler)  # needs to be in Qt context?
+        log.setLevel(self.log_level)
+        log.info("MultiPlotWindow connected to logging")
+
+    def open_experiment(self):
```

Comment:
I'm not a very big fan of having to amend these basic methods (i.e. `open_experiment` and `new_experiment`) of the `ManagedWindowBase` class, since this prohibits users to make there own ManagedWindow variant by simply add different TabWidgets. (additionally, at this moment some things wont work if you add another 'normal' PlotWidget.)

I think that everything should be handled by the widgets (or maybe the curves) itself, or maybe (but only when necessary of course, so I'm typically a bit reluctant on this regard) by generalising the definitions in the ManagedWindowBase class.